_analyzers/token-filters/common_gram.md

```diff
@@ -0,0 +1,165 @@
+---
+layout: default
+title: common_grams
+parent: Token filters
+nav_order: 60
+---
+<!-- vale off -->
+# Common_grams token filter
+<!-- vale on -->
+The `common_grams` token filter in OpenSearch improves search relevance by keeping commonly occurring phrases (common grams) in the text. This is useful when dealing with languages or datasets where certain word combinations frequently occur and can impact the search relevance if treated as separate tokens.
+
+Using this token filter improves search relevance by keeping common phrases intact, it can help in matching queries more accurately, particularly for frequent word combinations. It also improves search precision by reducing the number of irrelevant matches.
+
+Using this filter requires careful selection and maintenance of the list of common words
+{: .warning}
+
+## Parameters
+
+`common_grams` token filter can be configured with several parameters to control its behavior.
+
+Parameter | Description | Example
+`common_words` | A list of words that should be considered as common words. These words will be used to form common grams. (Required) | ["the", "and", "of"]
+`ignore_case` | Indicates whether the filter should ignore case differences when matching common words. | `true` or `false` (Default: `false`)
+`query_mode` | When set to true, the filter only emits common grams during the analysis phase (useful during query time to ensure the query matches documents analyzed with the same filter). | `true` or `false` (Default: `false`)
```

Comment:
When query_mode is set, the token output follows these guidelines:

  -   unigrams that are common_words are not included in the output.

 -    bigrams where a non-common word is followed by a common_word are retained in the output.

  -   unigrams of non-common words are excluded if they're immediately followed by a common_word.

  -   If a non-common word appears at the end of the text and is preceded by a common word, its unigram is also not included in the output.